### DISCUSSION TITLE
Multiple subscriptions and unsubscriptions handling

### DISCUSSION THREAD

**Discussion by MacherelR** [FreeOpcUa/opcua-asyncio#1476] [Q&A] [answered]
Hi everyone, 
I'm currently implementing some kind of gateway fetching data from an opc-ua server. To do so, I'm able to select through an HMI the nodes I want to subscribe to. I've already read that it is possible to subscribe to multiple nodes at once, but my question is:
Is there any, way to add a subscription to the actual subscription ? As example, let's say I launched a subscription for node ns=2;i=758 and it works well. Later on, I'd like to add a subscription for ns=2;i=90. Actually what I'm doing is running again this part of code:
```
    found_node = self.client.get_node(nodeid=self.node_id)
    found_node_name = await found_node.read_browse_name()
    self.node_map[self.node_id] = found_node_name.Name
    handler = SubHandler(self.node_map, producer=self.producer)
    self.sub = await self.client.create_subscription(period=1000,handler=handler)
    node = self.client.get_node(self.node_id)
    self.monitored_item = await self.sub.subscribe_data_change(node)
```
This happens in a custom implemented class called OPCUaSubscriber, and a new one is created at each subscription request. (only specificity is that they (subscribers) all share the same client instance which is passed as parameter when creating it)

Could anyone enlighten me whether my method is the right one or whether I could simplify ?

Thanks in advance

Rémy

**Comment by schroeder-** [comment, marked as answer]
You can just add new nodes to your subscription, when ever you want. You don't create a new subscription, just call `subscribe_data_change` to the new node.
```
    found_node = self.client.get_node(nodeid=self.node_id)
    found_node_name = await found_node.read_browse_name()
    self.node_map[self.node_id] = found_node_name.Name
    handler = SubHandler(self.node_map, producer=self.producer)
    self.sub = await self.client.create_subscription(period=1000,handler=handler)
    node = self.client.get_node(self.node_id)
    self.monitored_item = await self.sub.subscribe_data_change(node)
    .... 
    # do things
    new_node = self.client.get_node(self.new_node_id)
    await self.sub.subscribe_data_change(new_node)
```

**Reply by MacherelR** [reply]
Alright I'd told myself that something wasn't feeling right thanks a lot for your answer !
